roboteq_driver/src/controller.cpp

```diff
@@ -102,21 +101,21 @@ void Controller::read() {
       last_response_ = msg;
       last_response_available_.notify_one();
     } else if (msg[0] == '&') {
+      receiving_script_messages_ = true;
```

Comment:
Moving this here means that it doesn't get cleared right after `processStatus`, which is important since a script download can be triggered from there.